src/transformStateWithClones.js

```diff
@@ -8,6 +8,35 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const prevVersion = [];
+  const output = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(output, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete output[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in output) {
+          delete output[key];
+        }
+        break;
+
+      default:
+        return null;
```

Comment:
default case often used to catch errors, so it's better to provide some descriptive outcome
at least `return 'not supported action provided'`